Storage/SQL/SQLiteBookmarksSyncing.swift

```diff
@@ -646,6 +654,22 @@ extension SQLiteBookmarks {
         return self.db.queryReturnsNoResults("SELECT 1 FROM \(TableBookmarksLocal)")
     }
 
+    public func getLocalBookmarksAdditions() -> Deferred<Maybe<[GUID: BookmarkMirrorItem]>> {
+        let sql =
+        "SELECT * FROM \(TableBookmarksLocal) AS bookmarks " +
+        "WHERE type = \(BookmarkNodeType.bookmark.rawValue) AND sync_status = \(SyncStatus.new.rawValue) AND parentID = ?"
+        let args: Args = [BookmarkRoots.MobileFolderGUID]
+
+        return db.runQuery(sql, args: args, factory: BookmarkFactory.mirrorItemFactory) >>== { cursor in
+            var dict = [GUID: BookmarkMirrorItem]()
```

Comment:
We know already that the GUID is unique. You never actually use dictionary lookup; you only ever work with all the GUIDs or all the `BookmarkMirrorItem`s. I suggest you skip this dictifying: just return the array with `cursor.asArray()`.